### PR TITLE
Front end

### PR DESCRIPTION
**Pull request by omzmarlon**
This pull request include:
 - Basic Authentication of our web app (required some minor backend change).
 - We can now use protobuf in typescript (see README.md, there's still some further work to make code generation more autonomous)
 (Our web app does not support any error handling yet. so there may have some unexpected behaviour)

### CODE REVIEW

**Comment by tomyang729** [comment]
辛苦了💦  我这两天花时间看一下

File: play/src/main/scala/com/managementapp/services/common/ProtobufUtil.scala [hunk 1]
```diff
@@ -13,4 +13,9 @@ trait ProtobufUtil {
     JsonFormat.parser().merge(json, builder)
     builder.build.asInstanceOf[T]
   }
+
+  def generateJSON[M <: Message](content: M): String = {
```
Comment (tomyang729) [inline play/src/main/scala/com/managementapp/services/common/ProtobufUtil.scala]: The func name is a bit confusing the first time I saw it. Maybe pbToJson() is clearer? And the same for the one above. (jsonToPb() to keep them in consistent)

File: web/src/app-container/authentication/services/authenticationService.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import {TokenDTO} from "../../../dto/Authentication";
+
+@Injectable()
+export class AuthenticationService {
+    private static STORAGE_KEY: string = "token";
+
+    constructor(private _http: MAHttp) {}
+
+    public isUserLoggedIn(): boolean {
+        return (localStorage.getItem(AuthenticationService.STORAGE_KEY) !== null)
```
Comment (tomyang729) [inline web/src/app-container/authentication/services/authenticationService.ts]: Should we use cookie to store token? Localstoreage is generally not safe for sensitive data

File: play/src/main/scala/com/managementapp/services/common/ProtobufUtil.scala (hunk 1 above)
Comment (omzmarlon) [reply play/src/main/scala/com/managementapp/services/common/ProtobufUtil.scala]: Agree

File: web/src/app-container/authentication/services/authenticationService.ts (hunk 2 above)
Comment (omzmarlon) [reply web/src/app-container/authentication/services/authenticationService.ts]: True. And using cookies actually saves quite some code

**Comment by omzmarlon** [comment]
@tomyang729 Updated code based on review.
Renamed method in ProtobufUtil
And I switched to use cookies for authentication.
I didn't realize play has some nice support for cookie manipulation. Using cookies actually makes front-end and backend code cleaner.
And also don't need an "TokenDTO" anymore (we can just store username in cookies)

Reference: 
https://www.playframework.com/documentation/2.5.x/ScalaResults
http://alvinalexander.com/source-code/scala/how-getread-cookie-play-framework-2

**Review by tomyang729 — APPROVED** [review]

**Comment by tomyang729** [comment]
👍 🏅

**Pull request merged**